stdlib/tests/collections/mmr.rs

```diff
@@ -563,3 +619,102 @@ fn test_mmr_pack() {
     let advice_data = process.advice_provider.map.get(&hash_u8).unwrap();
     assert_eq!(stack_to_ints(advice_data), stack_to_ints(&expect_data));
 }
+
+#[test]
+fn test_mmr_add_single() {
+    let mmr_ptr = 1000;
+    let source = format!(
+        "
+        use.std::collections::mmr
+
+        begin
+            push.{mmr_ptr} # the address of the mmr
+            push.1.2.3.4   # the new peak
+            exec.mmr::add  # add the element
+        end
+    "
+    );
+
+    // when there is a single element, there is nothing to merge with, so the data is just in the
+    // MMR
+    #[rustfmt::skip]
+    let expect_data = &[
+        1, 0, 0, 0, // num_leaves
+        1, 2, 3, 4, // peak
+    ];
+    build_test!(&source).expect_stack_and_memory(&[], mmr_ptr, expect_data);
+}
+
+#[test]
+fn test_mmr_two() {
+    let mmr_ptr = 1000;
+    let source = format!(
+        "
+        use.std::collections::mmr
+
+        begin
```

Comment:
note: these values are _not_ in the merkle store, only their parent node

Reply:
one thing that i should have noted here, the children are not in the VM's memory, but they are in the VM's advice map, so the issue here is more of usability.